src/rust/engine/src/nodes.rs

```diff
@@ -943,10 +943,12 @@ impl Task {
                 Some(ty) if externs::is_union(ty) => {
                   let value = externs::get_value_from_type_id(ty);
                   let description = externs::project_str(&value, "union_description");
+                  let address_spec = externs::project_str(
```

Comment:
This isn't safe, because this will not always be a target.

Reply:
Don't unions always apply to target types?

Reply:
I guess one way around this is to have an (optional) "format_end_user_error_message" method on the @union that takes the subject as an argument? The @union does know what all its members have in common, presumably.

Not sure how to call python methods through extern though.

Reply:
Looks like extern::call_method is a good place to start